### PR TITLE
[WS-Passive-Scan] Base64 disclosure script with javascript

### PR DESCRIPTION
**Pull request by ManosMagnus**
TODOs
- [x] Add help Content 
- [x] Based on  #2089

### CODE REVIEW

**Comment by ManosMagnus** [comment]
Ready for review

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html [hunk 1]
```diff
@@ -11,5 +11,28 @@ <H1>WebSocket Passive Scan Rules</H1>
         <h2 id="scripts">Scripts</h2>
         Scripts which are included by default in the add-on and they implement the following WebSocket passive scan rules:
 
+        <h3>Base64 Disclosure</h3>
+
+        This script analyzes incoming WebSocket message payload for Base64 strings. The encoded information may include sensitive data which they may not specifically meant for end user consumption. In addition, those encoded payloads may include file signatures known as Magic Bytes. <br>
```
Comment (thc202) [inline addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: I think the `they` could be removed. I'd leave out the magic bytes reference, in itself is not a problem.

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html [hunk 2]
```diff
@@ -11,5 +11,28 @@ <H1>WebSocket Passive Scan Rules</H1>
...
+
+        The regular expression which is used to identify the Base64 string is based on padding characters. As result False Negative may be occured if the encoded string has length N bits where (N)mod6 = 0. On the other hand, False Positives may accured if the input text contains strings ending with '='. <br>
```
Comment (thc202) [inline addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: `may occur` (in both places).

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+base64Decoder = java.util.Base64.getDecoder();
+JavaString = Java.type("java.lang.String");
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+    var message = String(msg.getReadablePayload()).valueOf();
```
Comment (thc202) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: The `valueOf()` can be removed?

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            if(PRINT_RESULTS){
+                print("Message: " + message);
+                print("Evidence: " + evidence);
+                print("Decoded Evidence: " + decodedEvidence);
+            }
+
+            helper.newAlert()
+                .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+                .setName("Base64 Disclosure in WebSocket message (script)")
+                .setDescription("A Base64-encoded string has been founded in the websocket incoming message. Base64-encoded data may contain sensitive " +
```
Comment (thc202) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: `found`.

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                                "information such as usernames, passwords or cookies which should be further inspected. Decode evidence: "
```
Comment (thc202) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: `Decoded`.

**Comment by thc202** [comment]
The script should be renamed.

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html (hunk 1 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: Fixed f87355442

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html (hunk 2 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: Fixed in f87355442

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js (hunk 3 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: Fixed in 41aa3eeaf

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js (hunk 4 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: Fixed in 41aa3eeaf

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js (hunk 5 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/base64_disclosure.js]: Fixed in 41aa3eeaf

**Comment by ManosMagnus** [comment]
Script is renamed and PR is rebased

**Comment by thc202** [comment]
Thanks!

**Comment by ManosMagnus** [comment]
Thank you! Commits are squashed

**Pull request merged**